src/ammo_effect.cpp

```diff
@@ -6,53 +6,52 @@
 #include "generic_factory.h"
 #include "json.h"
 
-namespace
+generic_factory<ammo_effect> *get_all_ammo_effects()
```

Comment:
```suggestion
generic_factory<ammo_effect> &get_all_ammo_effects()
```
Why not return a reference?

Reply:
What difference does that make here? akrieger suggested the same thing so I must be missing something.

Reply:
It is better to return a reference if the return value can never be null, so the caller does not have to worry about dereferencing a null pointer. It also allows you to use the return value in the same way as a dereferenced value, so you don't have to replace `.` with `->`, use `*` to dereference it, or use `&` to get the address when returning.

Reply:
Alright, thanks.